File: solutions/Implement-strStr()/Implement-strStr().py


class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return a string or None
    def strStr(self, haystack, needle):
        if len(needle)==0:
            return haystack
        for i in range(len(haystack)-len(needle)+1):
            flag=True
            for j in range(len(needle)):
                if needle[j]!=haystack[i+j]:
                    flag=False
                    break
            if flag:
                return haystack[i:]
        return None
 

File: solutions/Implement-strStr()/Implement-strStr().js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  for (let i = 0; ; i++) {
    for (let j = 0; ; j++) {
      if (j === needle.length) {
        return i;
      }
      if (i + j === haystack.length) {
        return -1;
      }
      if (needle[j] !== haystack[i + j]) {
        break;
      }
    }
  }
};